.github/actions/build/action.yaml

```diff
@@ -60,3 +60,17 @@ runs:
           path: |
             src/out/${{ matrix.platform }}_qa/apks/*.apk
             src/out/${{ matrix.platform }}_qa/*_apk/*.apk
+      - name: Create Test Files Archive
+        if: startsWith(matrix.platform, 'linux') && matrix.config == 'devel'
+        run: |
+          set -x
+          cd src
+          tar cvf - out/${{ matrix.platform }}_${{ matrix.config }}/*tests out/${{ matrix.platform }}_${{ matrix.config }}/*.so out/${{ matrix.platform }}_${{ matrix.config }}/*.so.* out/${{ matrix.platform }}_${{ matrix.config }}/starboard/*.so.* | xz -T0 -1 -z - > test_artifacts.tar.xz
+        shell: bash
+      - name: Upload Test artifacts
+        if: startsWith(matrix.platform, 'linux') && matrix.config == 'devel'
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ inputs.test_artifacts_key }}
+          path: src/test_artifacts.tar.xz
+          retention-days: 3
```

Comment:
`IS_POSTSUBMIT` and `TESTING_ENABLED` was used for uploading to datadog. Can you add that step back in and guard it with those two env variables? Also add `continue-on-error: true` for this step so it doesn't affect the job status.

It should work once merged and is an important piece of this that would be nice to have from day one.